### ISSUE TITLE
Gradle 7.x support

### ISSUE THREAD

**Issue by jvmlet** [i-net-software/SetupBuilder#103]
Please support `7.x` line

**Comment by jvmlet** [comment]
Please merge  #104 
JPP has to be tweaked  since internal API  has a lot pf changes ….

**Comment by Horcrux7** [comment]
Which Gradle version do you want use? There seems API differences between 7.0, 7.1 and 7.2. Will 7.2 ok for you?

**Comment by jvmlet** [comment]
Yes, 7.2 will be great.

**Comment by jvmlet** [comment]
May I also suggest *NOT* to use gradle internal  API and instead - create branch for gradle versions  with breaking public API changes. This approach will also relieve you from JPP which is awkward...

**Comment by Horcrux7** [comment]
A not using of the internal API would change the DSL of SetupBuilder. The internal API is using for a Gradle like definition of the source files. A reimplementing of this DSL with own API would require a large amount of code. And it would break many Gradle scripts that use SetupBuilder.

Looking back it might have been better. But now it is too late.

**Comment by jvmlet** [comment]
> A not using of the internal API would change the DSL of SetupBuilder. The internal API is using for a Gradle like definition of the source files. A reimplementing of this DSL with own API would require a large amount of code. And it would break many Gradle scripts that use SetupBuilder.
> 
> Looking back it might have been better. But now it is too late.

You can cut the robe now and declare the starting from  ``

**Comment by jvmlet** [comment]
And I don't think that with current `copy ` spec API you will need a lot of code to allow source definition.. Just expose ` with copy spec`  from standard gradle API.

**Comment by jvmlet** [comment]
Hi, I had a chance to test your latest changes today. I've compiled the plugin locally and used `msi` task in our project we are migration to gradle `7.2` - the msi was correctly built, thanks.
When are you going to release  `7.2.x` line ?

**Comment by Horcrux7** [comment]
Version 7.2.11 is released.

**Comment by jvmlet** [comment]
Thanks @Horcrux7  and  I hope the upcoming gradle 8.0 with java 17 support will be compatible with `7.2.11`

**Comment by jvmlet** [comment]
Getting 
```
Input property 'i18n' with value '[com.inet.gradle.setup.msi.MsiLocalizedResource@1e405bc7]' cannot be serialized.
```
to reproduce- add `i18n` resource  :

```
msi{
   
    i18n {
        locale = "en-US"
        resource = file("windows/my-custom_en-US.properties").absolutePath
        overridable = true
    }
}
```
Please make  `MsiLocalizedResource` serializable

**Comment by jvmlet** [comment]
@Horcrux7 , please merge #106 , it  fixes the error

**Comment by jvmlet** [comment]
@Horcrux7 , would you please release the version with fix from #106 , we are stuck with `msi` task ....
